boards/FSM/Src/main.c

```diff
@@ -560,13 +558,13 @@ static void MX_TIM1_Init(void)
 
     /* USER CODE END TIM1_Init 1 */
     htim1.Instance               = TIM1;
-    htim1.Init.Prescaler         = 0;
+    htim1.Init.Prescaler         = 1;
     htim1.Init.CounterMode       = TIM_COUNTERMODE_UP;
     htim1.Init.Period            = TIM1_AUTO_RELOAD_REG;
     htim1.Init.ClockDivision     = TIM_CLOCKDIVISION_DIV1;
     htim1.Init.RepetitionCounter = 0;
     htim1.Init.AutoReloadPreload = TIM_AUTORELOAD_PRELOAD_DISABLE;
-    sConfig.EncoderMode          = TIM_ENCODERMODE_TI12;
+    sConfig.EncoderMode          = TIM_ENCODERMODE_TI1;
```

Comment:
noticed that the encoder mode here is changed, what does this change represent?

Reply:
TI12 counts up on rising edge on TIM2_CH1 or TIM2_CH2
TI1 counts up only on rising edge on TIM2_CH1
Felt like the latter made more sense and would be less noise-sensitive

Reply:
Just to be complete, I think TI1 means to count up/down on the rising/falling edge of TI1 based on the value of TI2.